boot.go

```diff
@@ -82,8 +82,12 @@ func main() {
 				log.Printf("Starting health check server on port %d", cnf.HealthSrvPort)
 				healthSrvCh := make(chan error)
 				go func() {
-					if err := healthsrv.Start(cnf.HealthSrvPort, kubeClient.Namespaces(), storageDriver, circ); err != nil {
-						healthSrvCh <- err
+					for{
+						if err := healthsrv.Start(cnf.HealthSrvPort, kubeClient.Namespaces(), storageDriver, circ); err != nil {
+							healthSrvCh <- err
+							break
+						}
+						log.Print("namespace watcher channel closed.Starting the watch again")
```

Comment:
that would mean we wouldn't be cleaning the apps until the builder has restarted.....currently on my cluster the channel is getting closed in less than an hour which would mean we would be cleaning only during the first hour